src/cbor.c

```diff
@@ -112,6 +126,16 @@ size_t cbor_writeToken(uint8_t type, uint64_t value, uint8_t* buffer, size_t buf
 
 	// Check sanity
 	switch (type) {
+	case CBOR_TYPE_NEGATIVE: {
+		int64_t negativeValue;
+		// reinterpret an actually negative value hidden in an unsigned in the safe way
+		memcpy(&negativeValue, &value, SIZEOF(value));
```

Comment:
> I guess we want some STATIC_ASSERT before this, e.g. SIZEOF(value) == SIZEOF(negativeValue).

I guess we could trust that 64_t types are the same size, but someone can switch it under us, so yeah, I'll add it. Doesn't cost us application size.